main.py

```diff
@@ -279,10 +287,40 @@ def run(play):
                     player["taken items"].append(map_location)
         if map["point" + str(map_location)]["enemy"] > 0 and map_location not in player["defeated enemies"]:
             enemies_remaining = map["point" + str(map_location)]["enemy"]
+            already_encountered = False
             while enemies_remaining > 0:
-                battle.fight(player, item)
+                battle.get_enemy_stats(player, item, enemy, map, map_location)
+                if not already_encountered:
+                    battle.encounter_text_show(player, item, enemy, map, map_location, enemies_remaining)
+                    already_encountered = True
+                battle.fight(player, item, enemy, map, map_location, enemies_remaining)
                 enemies_remaining -= 1
+            if round(random.uniform(.20, .50), 2) > .35:
```

Comment:
Wait—why is this an `if` statement? I think it might cause an error because of line 305 needing `list_enemies`.

Reply:
It has been removed in commit a52e654b07a9f013382a4d74f753b9fb6e83dec5